### ISSUE TITLE
learning all words does not increment each lifeform's "words learned" counter

### ISSUE THREAD

**Issue by wkitty42** [goatfungus/NMSSaveEditor#542]
tldr; when you use the editor to learn some or all of the words, the overall "Words Collected" property changes properly but the per lifeform "Words Learned" property does not get changed properly...

so in a normal game, you use a knowledge stone or talk to an entity... you learn a word, it gets checked off the list, and the "Words Collected" counter is incremented... you also gain +1 in the "Words Learned" count of the lifeform that speaks the language the word belongs to... you can see this in the Milestones tab when playing in a non-expedition game... this tab is replaced by the Expedition tab when in an expedition save...

in the Milestones tab, you can see the "Words Collected" count... now, in that same tab, select the Lifeforms on the left... you'll see they each have a "Words Learned" count... this is the count that gets the +1 i mention above... in the save editor, learning all the words does not increment these other counts... you can edit these counts in the raw JSON but first you need to figure out how many words there are in each language...

ideally, when manually checking or unchecking words or using the quick learn/unlearn buttons, each of the lifeforms' "Words Learned" counters will be adjusted up/down as well as the overall "Words Collected" counter being adjusted...

**Comment by AlHollandiyah** [comment]
I noticed the same! As a workaround you can uncheck 1 word for each of the 3 languages. Then you will only need to learn 1 word for each race from a life form or knowledge stone to have the correct number showing.

**Comment by wkitty42** [comment]
On 4/18/22 5:35 AM, Ayla Al-Hollandiyah wrote:
> I noticed the same! As a workaround you can uncheck 1 word for each of the 3 
> languages. Then you will only need to learn 1 word for each race from a life 
> form or knowledge stone to have the correct number showing.

i hadn't thought about that but i did check the numbers like that... i ran into 
this in one of my saves that i had edited... it now has the translator that 
needs to be calibrated and that calibration can't be done if one can't learn new 
words... i forget how many words it needs but i'll play with it...

thanks for that workaround... i'm going to leave this issue open in the hope 
that it can be fixed in the editor so we don't have to fumble about ;)

**Comment by ShadowSingularity** [comment]
Bit late (havent played in 6 years) but found an easy workaround for this, click the "learn all" and uncheck 3 words from the list (i unchecked the same one for gek, korvax, vykeen), go ingame and talk to some npc's to learn the words and the milestone should pop up, popped after 2 for me.
